src/descriptor/mod.rs

```diff
@@ -74,29 +75,43 @@ impl<Pk: MiniscriptKey> Descriptor<Pk> {
         &self,
         mut translatefpk: Fpk,
         mut translatefpkh: Fpkh,
-    ) -> Result<Descriptor<Q>, E>
+    ) -> Result<Result<Descriptor<Q>, Error>, E>
```

Comment:
I think we should do `Result<Result<Descriptor<Q>, E>, Error>` instead

Reply:
Actually, after some discussion on IRC, I think we should panic on uncompressed keys, and put a clear warning in the doccomment for this function.

If the user does not want to panic, this is totally in her control -- she can change `translatefpk` and `translatefpkh` to return an error rather than an uncompressed key.